service/src/main/java/skills/auth/saml/SAML2SecurityConfiguration.groovy

```diff
@@ -0,0 +1,111 @@
+import org.springframework.security.saml2.provider.service.authentication.OpenSaml4AuthenticationProvider;
+import org.springframework.security.saml2.provider.service.authentication.Saml2Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.saml2.provider.service.authentication.Saml2AuthenticatedPrincipal;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.springframework.stereotype.Component
+import skills.auth.PortalWebSecurityHelper
+import skills.auth.SecurityMode
+import skills.auth.UserInfo
+
+@Slf4j
+@Conditional(SecurityMode.SAML2Auth)
+@Component
+@Configuration
+@EnableWebSecurity
+/** Supports Single Identity Provider Only **/
+class SAML2SecurityConfiguration{
+
+    public static final String SKILLS_REDIRECT_URI = 'skillsRedirectUri'
+
+    @Value('${spring.security.saml2.metadata-location}')
+    String assertingPartyMetadataLocation;
+
+    @Value('#{"${spring.security.saml2.registrationId:okta-saml}"}')
+    String registrationId;
+
+    @Autowired
+    PortalWebSecurityHelper portalWebSecurityHelper
+
+    @Autowired
+    private RelyingPartyRegistrationRepository relyingPartyRegistrationRepository;
+
+    @Autowired
+    SAML2Utils saml2Utils;
+
+
+     @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception   {
+
+         OpenSaml4AuthenticationProvider authenticationProvider = new OpenSaml4AuthenticationProvider();
+         authenticationProvider.setResponseAuthenticationConverter(responseToken -> {
+             Saml2Authentication authentication = OpenSaml4AuthenticationProvider
+                     .createDefaultResponseAuthenticationConverter()
+                     .convert(responseToken);
+             Saml2AuthenticatedPrincipal principal = (Saml2AuthenticatedPrincipal) authentication.getPrincipal();
+            // Assertion assertion = responseToken.getResponse().getAssertions().get(0);
+             UserInfo userInfo = saml2Utils.convertToUserInfo(authentication,true);
+
+             return new Saml2Authentication(principal,authentication.getSaml2Response(),userInfo.getAuthorities());
+         });
+
+         portalWebSecurityHelper.configureHttpSecurity(http);
+         http.saml2Login(saml2 -> saml2
+                 .authenticationManager(new ProviderManager(authenticationProvider))
+                 .relyingPartyRegistrationRepository(relyingPartyRegistrationRepository())
+         );
+         http.logout((logout) -> logout.logoutSuccessUrl("/local/logout"));
+         return http.build();
+    }
+    @Bean
+    RelyingPartyRegistrationRepository relyingPartyRegistrationRepository() {
+        RelyingPartyRegistration registration = RelyingPartyRegistrations.fromMetadataLocation(assertingPartyMetadataLocation)
+                .registrationId(registrationId)
+                .build()
+        return new InMemoryRelyingPartyRegistrationRepository(registration);
```

Comment:
noticed that `SAML2SecurityConfiguration.groovy` uses `InMemoryRelyingPartyRegistrationRepository`. Will this work in a multi-node installation with multiple skills-service daemons behind a load balancer, or should we consider alternative strategies like storing sessions in Postgres to support distributed instances?

Reply:
It should work as relyingPartyRegistrationRepository bean fetches the asserting party meta data from Identity provider